src/com/sap/piper/tools/JavaArchiveDescriptor.groovy

```diff
@@ -0,0 +1,62 @@
+package com.sap.piper.tools
+
+import com.sap.piper.EnvironmentUtils
+import com.sap.piper.FileUtils
+
+import hudson.AbortException
+
+
+class JavaArchiveDescriptor extends ToolDescriptor {
+
+    final javaTool
+    final javaOptions
+
+    JavaArchiveDescriptor(name, environmentKey, stepConfigurationKey, executablePath, executableName, version, versionOption, javaTool, javaOptions) {
+        super(name, environmentKey, stepConfigurationKey, executablePath, executableName, version, versionOption)
+        this.javaTool = javaTool
+        this.javaOptions = javaOptions
+    }
+
+    @Override
+    def getHome(script, configuration, log = true) {
+
+        def home
+        if (EnvironmentUtils.isEnvironmentVariable(script, environmentKey)) {
+            home = EnvironmentUtils.getEnvironmentVariable(script, environmentKey)
+            if (log) script.echo "$name home '$home' retrieved from environment."
+        }
+        else if (configuration.containsKey(stepConfigurationKey)) {
+            home = configuration.get(stepConfigurationKey)
+            if (log) script.echo "$name home '$home' retrieved from configuration."
+        } else if (isOnCurrentWorkingDirectory(script)){
+            home = ''
+            if (log) script.echo "$name expected on current working directory."
+        } else {
+            throw new AbortException(getConfigurationOptions())
+        }
+        return home
+    }
+
+    @Override
+    def getExecutable(script, configuration, log = true) {
+
+        def tool = getTool(script, configuration, log)
```

Comment:
Should we try to get a better name for ```tool``` here? It is basically the uber jar file which is triggered later on. We can say, well this is nit-picking and it is only about naming here. But on the other hand: naming is important, bad names will confuse our successors who have to maintain the code later (or even ourselfs in two or three years).